### PR TITLE
Add Typescript definitions

### PR DESCRIPTION
**Pull request by dlee-nvisia**
This pull request primarily adds a definitions file so that the library can be used easily in Typescript projects. There is a single change to the actual codebase - I've registered the CloudWatch transport on `winston.transports` in order to allow for idiomatic importing for Typescript projects.
`
import * as winston from 'winston';

import 'winston-cloudwatch';

winston.add(winston.transports.CloudWatch, {});
`

Please let me know if there are any problems with this or if the types of any variables can be improved.

### CODE REVIEW

**Comment by lazywithclass** [comment]
Thanks a lot for this!

I don't have any experience with TypeScript so please allow me to broaden my knowlege over the weekend. After that I hope to have more info on the matter, so that I can review it effectively.

File: index.js [hunk 1]
```diff
@@ -124,4 +124,6 @@ WinstonCloudWatch.prototype.kthxbye = function(callback) {
 
 function stringify(o) { return JSON.stringify(o, null, '  '); }
 
+winston.transports.CloudWatch = WinstonCloudWatch;
```
Comment (lazywithclass) [inline index.js]: I think it's already being done by Winston.
If you try to run one of the examples and look into `winston` you should see `WinstonCloudwatch`.

**Comment by lazywithclass** [comment]
@dlee-nvisia Besides the above comment I would say everything looks correct to me. As I said I'm not a TypeScript user but I'm open to to refining this if you have any other PRs in the future!

**Comment by lazywithclass** [comment]
👍 Ok I will merge this later on tonight in 1.10.0, thanks again for your effort.

**Comment by dlee-nvisia** [comment]
I actually just did a little more testing in my own project - and it turns out that line is required. If I remove it, I started to get `TypeError: transport is not a constructor` around the `.transport` property. Do you have any suggestions? Or should I just restore that line.

**Comment by lazywithclass** [comment]
I think that's something related to TypeScript, I don't have that error when I run one of the examples.
You could try creating an example in TypeScript and pasting it here, I am happy to give it a go myself also to help you debug the issue.

If we don't find a cause in a reasonable amount of time I am happy to add that line.

**Comment by dlee-nvisia** [comment]
So, I made a cut-down test test example. I can still reproduce the error. My gut feeling is it could be related to Strict Mode - my test case has that at the top, but tsc will add it to the generated js code even if you leave it off.

```'use strict';
import * as winston from 'winston';
import 'winston-cloudwatch';

const options: winston.CloudwatchTransportOptions = {
  level: (process.env.LOG_LEVEL || 'debug'),
  logGroupName: (process.env.ENVIRONMENT_NAME || 'local'),
  logStreamName: (process.env.LOG_STREAM_NAME || 'local'),
  awsRegion:  (process.env.AWS_REGION || 'us-west-2'),
};

// Verify transport is registered with the default logger
if (!winston['default'].transports.hasOwnProperty('CloudWatch')) {
  winston.add(winston.transports.CloudWatch, options);
}

winston.debug('Typescript Logging Test!');```

**Comment by lazywithclass** [comment]
Ok no worries, I will publish later on. Cheers.

**Comment by lazywithclass** [comment]
1.10.0 published, thank you again for your effort!

**Comment by lazywithclass** [comment]
@dlee-nvisia 1.10.0 published, thank you again for your effort!

**Pull request merged**